### PR TITLE
fixing default flag for hints autodiscovery

### PR DESCRIPTION
**Pull request by gizas**
- Enhancement

## What does this PR do?

Updates the default flag of hint configuration for container logs processing

## Why is it important?
Related to https://github.com/elastic/elastic-agent/pull/2981#issuecomment-1618493029

### CODE REVIEW

**Review by ChrsMark — APPROVED** [review]
lgtm

File: internal/pkg/composable/providers/kubernetes/pod.go [hunk 1]
```diff
@@ -447,7 +447,7 @@ func generateContainerData(
 							)
 						} else if config.Hints.DefaultContainerLogs {
 							// in case of no package detected in the hints fallback to the generic log collection
-							_, _ = hintsMapping.Put("generic_logs.container_logs.enabled", true)
+							_, _ = hintsMapping.Put("container_logs.enabled", true)
```
Comment (pchila) [inline internal/pkg/composable/providers/kubernetes/pod.go:450]: Changing the key of the mapping is ok but maybe we should be consistent as we have the same key in line 478...

Comment (ChrsMark) [reply internal/pkg/composable/providers/kubernetes/pod.go:450]: Ah good catch 🤦🏽 . @gizas could you also change https://github.com/elastic/beats/pull/35999/files#diff-050ad6dbacd48af0a47953cade64df8f81c2104c232ddc15dfc1feff29d48badR40-R43 accordingly?

**Review by ChrsMark — CHANGES_REQUESTED** [review]
We need to also tune the container's part as mentioned at https://github.com/elastic/elastic-agent/pull/2986#discussion_r1252980299

File: internal/pkg/composable/providers/kubernetes/pod.go (hunk 1 above)
Comment (gizas) [reply internal/pkg/composable/providers/kubernetes/pod.go:450]: Done. I also updated the relevant manifests

File: deploy/kubernetes/elastic-agent-kustomize/default/elastic-agent-standalone/base/elastic-agent-standalone-daemonset-configmap.yaml [hunk 2]
```diff
@@ -359,28 +359,26 @@ data:
               #     match: after
             paths:
               - /var/log/containers/*${kubernetes.container.id}.log
-      - name: filestream-generic
```
Comment (ChrsMark) [inline deploy/kubernetes/elastic-agent-kustomize/default/elastic-agent-standalone/base/elastic-agent-standalone-daemonset-configmap.yaml:362]: Those should also be remove completely with https://github.com/elastic/ingest-docs/pull/304.

File: deploy/kubernetes/elastic-agent-standalone-kubernetes.yaml [hunk 3]
```diff
@@ -359,28 +359,25 @@ data:
               #     match: after
             paths:
               - /var/log/containers/*${kubernetes.container.id}.log
-      - name: filestream-generic
-        id: hints-container-logs-${kubernetes.hints.container_id}
+      - name: hints-filestream-container-logs
+        id: hints-filestream-container-logs-${kubernetes.hints.container_id}
```
Comment (ChrsMark) [inline deploy/kubernetes/elastic-agent-standalone-kubernetes.yaml:363]: Won't be remove?

File: deploy/kubernetes/elastic-agent-kustomize/default/elastic-agent-standalone/base/elastic-agent-standalone-daemonset-configmap.yaml (hunk 2 above)
Comment (gizas) [reply deploy/kubernetes/elastic-agent-kustomize/default/elastic-agent-standalone/base/elastic-agent-standalone-daemonset-configmap.yaml:362]: Removed!

File: deploy/kubernetes/elastic-agent-standalone-kubernetes.yaml (hunk 3 above)
Comment (ChrsMark) [reply deploy/kubernetes/elastic-agent-standalone-kubernetes.yaml]: So does this mean that the Makefile is wrong? It looks weird to me why we can't just remove those sections.

File: deploy/kubernetes/elastic-agent-managed-kubernetes.yaml [hunk 4]
```diff
@@ -30,7 +30,7 @@ spec:
       dnsPolicy: ClusterFirstWithHostNet
       containers:
         - name: elastic-agent
-          image: docker.elastic.co/beats/elastic-agent:8.3.0
+          image: docker.elastic.co/beats/elastic-agent:8.8.1
```
Comment (ChrsMark) [inline deploy/kubernetes/elastic-agent-managed-kubernetes.yaml:33]: Is this expected?

File: deploy/kubernetes/elastic-agent-standalone-kubernetes.yaml [hunk 5]
```diff
@@ -712,7 +689,7 @@ spec:
       #        mountPath: /etc/elastic-agent/inputs.d
       containers:
         - name: elastic-agent-standalone
-          image: docker.elastic.co/beats/elastic-agent:8.3.0
+          image: docker.elastic.co/beats/elastic-agent:8.8.1
```
Comment (ChrsMark) [inline deploy/kubernetes/elastic-agent-standalone-kubernetes.yaml:692]: Is this expected?

File: deploy/kubernetes/elastic-agent-standalone-kubernetes.yaml [hunk 6]
```diff
@@ -840,6 +817,619 @@ spec:
         #    path: /var/cache/Elastic
         #    type: DirectoryOrCreate
 ---
+# Configmap for Daemoset Agents. It includes all the datastreams except Kube-State-metrics
```
Comment (ChrsMark) [inline deploy/kubernetes/elastic-agent-standalone-kubernetes.yaml:820]: Why is this added here? Looks weird and kind of irrelevant to the PR?

**Review by ChrsMark — CHANGES_REQUESTED** [review]
@gizas let's make sure we clean the changes in this PR. Specifically we need to see just the 2 following changes and nothing more here:

1. Remove the `generic_logs.*` prefix from the mapping creation (in provider's code)
2. remove the existent templates from the any k8s manifests that exist for standalone Agent.

File: deploy/kubernetes/elastic-agent-managed-kubernetes.yaml (hunk 4 above)
Comment (gizas) [reply deploy/kubernetes/elastic-agent-managed-kubernetes.yaml:33]: yes

File: deploy/kubernetes/elastic-agent-standalone-kubernetes.yaml (hunk 5 above)
Comment (gizas) [reply deploy/kubernetes/elastic-agent-standalone-kubernetes.yaml:692]: yes

File: deploy/kubernetes/elastic-agent-standalone-kubernetes.yaml (hunk 3 above)
Comment (gizas) [reply deploy/kubernetes/elastic-agent-standalone-kubernetes.yaml]: Well I had introduced them for Kustomize creation. I would say to keep them commented for now in case we want to update the ksutomize templates

File: deploy/kubernetes/elastic-agent-standalone-kubernetes.yaml (hunk 6 above)
Comment (gizas) [reply deploy/kubernetes/elastic-agent-standalone-kubernetes.yaml]: Removed by correcting the makefile and rerun

**Review by ChrsMark — APPROVED** [review]

**Review by pchila — APPROVED** [review]
